services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainMetrics.java

```diff
@@ -171,6 +208,19 @@ public void onSlot(final UInt64 slot) {
 
   private void updateMetrics(final StateAndBlockSummary head) {
     final BeaconState state = head.getState();
+    state
+        .getTransitionCaches()
+        .getLatestTotalBalances()
+        .ifPresent(
+            totalBalances -> {
+              currentEpochTotalWeight.set(totalBalances.getCurrentEpoch().longValue());
+              previousEpochTotalWeight.set(totalBalances.getPreviousEpoch().longValue());
+
+              currentEpochParticipationWeight.set(
+                  totalBalances.getCurrentEpochAttesters().longValue());
+              previousEpochParticipationWeight.set(
+                  totalBalances.getPreviousEpochAttesters().longValue());
```

Comment:
Now that I think about it, these numbers don't quite work out like this (sorry).  The `TotalBalances` are created at the end of the epoch so the "current" balance is actually "previous" by the time we're actually updating the metrics.  I suspect we should just drop the "current" version of these metrics as they're not particularly useful anyway.

Then we wind up setting:
```suggestion
              previousEpochTotalWeight.set(totalBalances.getCurrentEpoch().longValue());

              previousEpochParticipationWeight.set(
                  totalBalances.getCurrentEpochAttesters().longValue());
```

and add a comment explaining why it's right even though it looks weird. :) Also need to consider if we have to worry about overflowing the signed long value with these metrics.  Prometheus actually uses a `double` for everything so we could potentially convert it to ETH if there's a risk of overflow.

Reply:
I read that the double MAX value is: 1.7*10^308. So I don't think that will be an issue. Am I missing something?

Reply:
Put the comment and made the change for previous/current epoch confusion.